File: Python/test_harness.py
import sys

class TestHarness:
  'Basic (very basic) testing framework'
  nPass = 0
  nFail = 0
  nTotalPass = 0
  nTotalFail = 0
  nTestCases = 0

  @classmethod
  def Summary(self):
    print("Summary")
    print("  Passed: %d" % self.nPass)
    print("  Failed: %d\n" % self.nFail)

  @classmethod
  def FinalSummary(self):
    if self.nPass > 0 or self.nFail > 0:
      self.Summary()
    self.nTotalPass += self.nPass
    self.nTotalFail += self.nFail
    print("\nFinal Summary")
    print("  %5d Total Cases: " % self.nTestCases)
    print("  %5d Total Tests" % (self.nTotalPass + self.nTotalFail))
    print("  %5d Total Passed" % self.nTotalPass)
    print("  %5d Total Failed" % self.nTotalFail)

  @classmethod
  def TestCase(self,msg):
    if self.nPass > 0 or self.nFail > 0:
      self.Summary()
    self.nTotalPass += self.nPass
    self.nTotalFail += self.nFail
    self.nPass = 0
    self.nFail = 0
    self.nTestCases += 1
    print(msg)

  @classmethod
  def Test(self,msg,result):
    if result:
      self.nPass+=1
      print(" ",msg,"... pass")
    else:
      self.nFail+=1
      print(" ",msg,"... fail")

  @classmethod
  def RunTests(self,tests,test_numbers):
    nTests = len(tests)
    if len(test_numbers) > 0:
      for itest in test_numbers:
        i = int(itest)
        if i < nTests:
          tests[i]()
        else:
          print("No test for ",itest)
    else:
      for test in tests:
        test()
    self.FinalSummary()

File: Lua/TestHarness.lua

TestHarness = { nTestCases=0, nFail = 0, nPass = 0, nTotalPass=0,nTotalFail=0}
TestHarness.__index = TestHarness

function TestHarness:new()
  o = { }
  setmetatable(o,self)
  self.__index=self
  return o
end

function TestHarness:summary()
  print(string.format("Summary\n  Passed: %4d\n  Failed: %4d\n",self.nPass,self.nFail))
end

function TestHarness:testcase(s)
  self.nTestCases=self.nTestCases+1
  if(self.nPass > 0 or self.nFail > 0) then
    self:summary()
  end
  print(s)
  self.nTotalPass = self.nTotalPass + self.nPass
  self.nTotalFail = self.nTotalFail + self.nFail
  self.nPass = 0
  self.nFail = 0
end

function TestHarness:test(s,t)
  res=""
  if (t) then
    self.nPass= self.nPass + 1
    res="Pass"
  else
    self.nFail= self.nFail + 1
    res="Fail"
  end
  print(string.format("  %s .. %s",s,res))
end

function TestHarness:final_summary()
  if(self.nPass > 0 or self.nFail > 0) then
    self:summary()
  end
  self.nTotalPass = self.nTotalPass + self.nPass
  self.nTotalFail = self.nTotalFail + self.nFail
  print(string.format("\nFinal Summary\n %6d Test Cases\n %6d Total Tests\n %6d Total Passed\n %6d Total Failed",
      self.nTestCases,(self.nTotalPass+self.nTotalFail),self.nTotalPass,self.nTotalFail))
end
